From 05c8dcd26a56845bc5add5125908cc7d56b55ee5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ignacy=20Kuchci=C5=84ski?= <ignacykuchcinski@gmail.com>
Date: Tue, 17 Jun 2025 15:55:53 +0200
Subject: [PATCH 1/5] avahi: Build without gtk3 support

Currently we build avahi with gtk3 support, which is only needed for
avahi-discover gtk3 tool. This prevents depending on avahi when building
cups, because of circular dependency cups-base -> avahi -> gtk3 -> cups.

Cups ippfind utility will not build without avahi, so break the
dependency chain by dropping the gtk3 support in avahi.
---
 elements/components/avahi-base.bst | 6 +-----
 elements/components/avahi-ui.bst   | 8 --------
 2 files changed, 1 insertion(+), 13 deletions(-)

diff --git a/elements/components/avahi-base.bst b/elements/components/avahi-base.bst
index 36748826f..ca42d9a3e 100644
--- a/elements/components/avahi-base.bst
+++ b/elements/components/avahi-base.bst
@@ -5,7 +5,6 @@ build-depends:
 - public-stacks/buildsystem-autotools.bst
 - components/dbus.bst
 - components/glib.bst
-- components/gtk3.bst
 - components/systemd.bst
 
 - components/libdaemon.bst
@@ -21,7 +20,7 @@ variables:
     --disable-qt4
     --disable-qt5
     --disable-gtk
-    --enable-gtk3
+    --disable-gtk3
     --disable-python
     --disable-mono
 
@@ -50,10 +49,7 @@ public:
       - "%{bindir}/bssh"
       - "%{bindir}/bvnc"
       - "%{bindir}/bshell"
-      - "%{libdir}/libavahi-ui-gtk3.so*"
-      - "%{libdir}/pkgconfig/avahi-ui-gtk3.pc"
       - "%{debugdir}%{bindir}/bssh.debug"
-      - "%{debugdir}%{libdir}/libavahi-ui-gtk3.so*.debug"
       gobject:
       - "%{includedir}/avahi-gobject"
       - "%{includedir}/avahi-gobject/**"
diff --git a/elements/components/avahi-ui.bst b/elements/components/avahi-ui.bst
index bddf6d095..3b1a60161 100644
--- a/elements/components/avahi-ui.bst
+++ b/elements/components/avahi-ui.bst
@@ -5,16 +5,8 @@ build-depends:
 
 runtime-depends:
 - components/avahi.bst
-- components/gtk3.bst
 
 config:
   include:
   - ui
   include-orphans: False
-
-public:
-  bst:
-    split-rules:
-      devel:
-        (>):
-        - "%{libdir}/libavahi-ui-gtk3.so"
-- 
2.49.0

